content/course-section/calendar.md

```diff
@@ -0,0 +1,9 @@
+menu: 
+    main:
+        name: "Calendar"
```

Comment:
how does Hugo determine the path to this item? Or is that part of the template?

Reply:
The file is under `content\course-section\calendar.md`, so a page will be generated at http://localhost:1313/course-section/calendar/.  The menu declaration in each section's front matter tells hugo to generate a menu item in the menu named "main" with the name of whatever that section is.  If you look in the archetype for course-section, you'll see this:

`menu: 
    main:
        name: "{{ replace .Name "-" " " | title }}"`
This ensures that when you are creating a new "course-section" element that a menu item is generated in the main menu for that section.  It's iterated in the template (baseof.html) like so:

`{{ $currentPage := . }}
{{ range .Site.Menus.main }}
<li class="nav-item">
  <a class="nav-link{{if or ($currentPage.IsMenuCurrent "main" .) ($currentPage.HasMenuCurrent "main" .) }} active{{end}}" 
    href="{{ .URL }}" title="{{ .Title }}">
    {{ .Name }}
  </a>
</li>
{{ end }}`